src/uct/tcp/tcp_iface.c

```diff
@@ -369,6 +372,53 @@ static ucs_status_t uct_tcp_iface_listener_init(uct_tcp_iface_t *iface)
     return status;
 }
 
+static ucs_status_t uct_tcp_iface_set_ports_range(uct_tcp_iface_t *iface,
+                                                  uct_tcp_iface_config_t *config)
+{
+    ucs_range_spec_t system_ports_range;
+    unsigned start_range, end_range;
+    ucs_status_t status;
+
+    if ((config->ports_range.first == 0) && (config->ports_range.last == 0)) {
+        /* using a random port */
+        goto out_set_config;
+    }
+
+    /* get the system configuration for usable ports range */
+    status = ucs_sys_get_ip_local_port_range(&system_ports_range);
+    if (status != UCS_OK) {
+        /* using the user's config */
+        goto out_set_config;
+    }
+
+    /* find a common range between the user's ports range and the one on the system */
+    start_range = ucs_max(system_ports_range.first, config->ports_range.first);
+    end_range   = ucs_min(system_ports_range.last, config->ports_range.last);
+
+    if (start_range > end_range) {
+        /* there is no common range */
+        ucs_error("no common ports range. user's range: %d-%d, system range %d-%d",
+                  config->ports_range.first, config->ports_range.last,
+                  system_ports_range.first, system_ports_range.last);
+        status = UCS_ERR_OUT_OF_RANGE;
+        goto out;
+    }
+
+    iface->ports_range.first = start_range;
+    iface->ports_range.last  = end_range;
+
+    goto out_print;
+
+out_set_config:
+    status                   = UCS_OK;
+    iface->ports_range.first = config->ports_range.first;
+    iface->ports_range.last  = config->ports_range.last;
+out_print:
+    ucs_debug("using TCP port range: %d-%d", iface->ports_range.first, iface->ports_range.last);
+out:
+    return status;
```

Comment:
pls separate error from success so it will look consistent with other functions
e.g
```
    iface->port_range =  ...
    ucs_debug(...)
    return UCS_OK;
err:
    return status;
}
```